src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1971,4 +1980,7 @@
   <data name="UseSpanClearInsteadOfFillTitle" xml:space="preserve">
     <value>Prefer 'Clear' over 'Fill'</value>
   </data>
+  <data name="ImplementGenericMathInterfacesCorrectlyMessageForClass" xml:space="preserve">
+    <value>The '{0}' class requires the '{1}' type parameter to be filled with the derived type '{2}'</value>
+  </data>
```

Comment:
Could both "ForClass" and "ForInterface" be the same message, e.g:

> '{0}' requires the '{1}' type parameter to be filled with the derived type '{2}'

So that we can merge the two diagnostic descriptors into one?

Reply:
I was wondering if I should add another version for record, but integrating all into one message sounds better, wondering if I should use `type` for integrated message, I think `type` refers to all: `class, interface and record`, the message would become `'{0}' type requires the '{1}' type parameter to be filled with the derived type '{2}'`, but `type` used twice in this case maybe just use `'{0}' requires the '{1}' type parameter to be filled with the derived type '{2}'` instead. CC @eerhardt

Reply:
`'{0}' requires the '{1}' type parameter to be filled with the derived type '{2}'` is better